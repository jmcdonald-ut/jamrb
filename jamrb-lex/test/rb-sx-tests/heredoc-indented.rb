def foo
  standard = <<INDENTED_TEXT
Identents are considered part of the text like so.
  HEREDOC
INDENTED_TEXT

  indented = <<-INDENTED_HEREDOC
  This actually ignores the first ident, this may mean I should capture how
  deep we are idented?
  What happens with this tab?
  INDENTED_HEREDOC

  least_indented = <<~SQUIGGLY_HEREDOC
This will treat the least indented text as the beginning left column.
  why?
    SQUIGGLY_HEREDOC
end
