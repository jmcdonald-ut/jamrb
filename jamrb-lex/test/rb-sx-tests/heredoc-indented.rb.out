((Token ((1 0) on_kw "def"))
 (Token ((1 3) on_sp " "))
 (Token ((1 4) on_ident "foo"))
 (Token ((1 7) on_nl "\n"))
 (Token ((2 0) on_sp "  "))
 (Token ((2 2) on_ident "standard"))
 (Token ((2 10) on_sp " "))
 (Token ((2 11) on_op "="))
 (Token ((2 12) on_sp " "))
 (Token ((2 13) on_heredoc_beg "<<INDENTED_TEXT"))
 (Token ((2 28) on_nl "\n"))
 (Token
  ((3 0)
   on_tstring_content
   "Identents are considered part of the text like so.\n  HEREDOC\n"))
 (Token ((5 0) on_heredoc_end "INDENTED_TEXT\n"))
 (Token ((6 0) on_ignored_nl "\n"))
 (Token ((7 0) on_sp "  "))
 (Token ((7 2) on_ident "indented"))
 (Token ((7 10) on_sp " "))
 (Token ((7 11) on_op "="))
 (Token ((7 12) on_sp " "))
 (Token ((7 13) on_heredoc_beg "<<-INDENTED_HEREDOC"))
 (Token ((7 32) on_nl "\n"))
 (Token
  ((8 0)
   on_tstring_content
   "  This actually ignores the first ident, this may mean I should capture how\n  deep we are idented?\n  What happens with this tab?\n"))
 (Token ((11 0) on_heredoc_end "  INDENTED_HEREDOC\n"))
 (Token ((12 0) on_ignored_nl "\n"))
 (Token ((13 0) on_sp "  "))
 (Token ((13 2) on_ident "least_indented"))
 (Token ((13 16) on_sp " "))
 (Token ((13 17) on_op "="))
 (Token ((13 18) on_sp " "))
 (Token ((13 19) on_heredoc_beg "<<~SQUIGGLY_HEREDOC"))
 (Token ((13 38) on_nl "\n"))
 (Token
  ((14 0)
   on_tstring_content
   "This will treat the least indented text as the beginning left column.\n  why?\n"))
 (Token ((16 0) on_heredoc_end "    SQUIGGLY_HEREDOC\n"))
 (Token ((17 0) on_kw "end"))
 (Token ((17 3) on_nl "\n")))
